holidays/calendars/islamic.py

```
@@ -4102,5 +4104,110 @@ def tasua_dates(self, year: int) -> Iterable[tuple[date, bool]]:
+    def _ephem(self):
+        """Lazy-load and cache the ephem library."""
+        try:
+            import ephem
+
+            return ephem
+        except ImportError as exc:
+            raise ImportError(
+                "The 'ephem' library is required for MABIMS lunar calculations."
+            ) from exc
+
+    def _get_observer(self, dt: date):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_get_observer`

(ANN202)